.github/workflows/dotnet.yml

```diff
@@ -270,6 +270,35 @@ jobs:
     - name: Checkout code again
       uses: actions/checkout@v3
 
+    # Update file versions.
+    - name: Update file versions
+      run: |
+        # Define the new version number
+        $version = $env:GITHUB_ENV['needs.create_release.outputs.tag_name']
+
+        # Get all non-test .csproj files in the solution directory and its subdirectories
+        $csprojFiles = Get-ChildItem -Path $env:GITHUB_WORKSPACE -Recurse -Filter "*.csproj"
+          | Where-Object { $_.Name -notmatch "test" }
+
+        foreach ($file in $csprojFiles) {
+          # Read the content of the .csproj file
+          $content = Get-Content -Path $file.FullName
+
+          # Update the version number in the .csproj file
+          $updatedContent = $content -replace '<FileVersion>.*?</FileVersion>', "<FileVersion>$newVersion</FileVersion>"

```

Comment:
Should this be $version?

Reply:
When I tried with $version it got interpreted as nuget version. I tested with both and FileVersion is the correct property. But I'm going to try setting it from the publish command so I don't have to modify the project files.